app/src/main/java/com/oracle/javafx/scenebuilder/app/SceneBuilderApp.java

```diff
@@ -656,12 +746,15 @@ public DocumentWindowController getFrontDocumentWindow() {
         return null;
     }
 
-    private void performOpenFiles(List<File> fxmlFiles) {
+    private FileOpenResult performOpenFiles(List<File> fxmlFiles) {
```

Comment:
My last comment to update `performOpenFiles` was to update this method instead. Since `handleFileOpenResult` only does the following:

```
        if (result.hasErrors()) {
            showFileOpenErrors(result, owner);
        } else {
            onSuccess.run();
        }
```

it should be possible to incorporate it in the same method by changing it to `void performOpenFiles(final List<File> files, Runnable onSuccess)`. This way we won't need `FileOpenResult` record as well.

Or, do you see a potential upside of having the record and the additional methods to handle everything?

Reply:
The potential I see is separation of concerns and with that the potential to get this all testable.
The reason why I chose this way was to separate UI actions, file handling and error handling, so that we might later implement tests. Also (for later) when drag&drop can be used, we would have a result type which bundles the file to be opened with a potential exception which occured.

This way one could even move the performOpenFiles logic into a separate class and make the SceneBuilderApp class a little smaller. That would move the design more towards building blocks.

Reply:
I'll try to consolidate the behavior and implement a test for the file open logic accordingly.

Reply:
> The reason why I chose this way was to separate UI actions, file handling and error handling, so that we might later implement tests.

We can file a separate issue and work on it in a PR of its own.

Reply:
I agree with that approach.